packages/core/src/global.js

```diff
@@ -119,14 +111,14 @@ export let Global: React.AbstractComponent<
         // insert keyframes
         insertStyles(cache, serialized.next, true)
       }
+
       let sheet: StyleSheet = ((sheetRef.current: any): StyleSheet)
-      if (sheet.tags.length) {
-        // if this doesn't exist then it will be null so the style element will be appended
-        let element = sheet.tags[sheet.tags.length - 1].nextElementSibling
-        sheet.before = ((element: any): Element | null)
+      sheet.before = cache.sheet.tags.length ? cache.sheet.tags[0] : null
+      cache.insert(``, serialized, sheet, false)
+
+      return () => {
```

Comment:
refactored those 2 hooks a little bit, as discussed briefly on Slack, an extra careful review is advised here